packages/logger/src/node.ts

```diff
@@ -0,0 +1,159 @@
+};
+
+/**
+ * Setup a CLI logger, common for beacon, validator and dev commands
+ */
+export function getNodeLogger(opts: LoggerNodeOpts): LoggerNode {
+  return WinstonLoggerNode.fromNewTransports(opts);
+}
+
+function getNodeLoggerTransports(opts: LoggerNodeOpts): winston.transport[] {
+  const consoleTransport = new ConsoleDynamicLevel({
+    // Set defaultLevel, not level for dynamic level setting of ConsoleDynamicLvevel
+    defaultLevel: opts.level,
+    debugStdout: true,
+    handleExceptions: true,
+  });
+
+  if (opts.levelModule) {
+    for (const [module, level] of Object.entries(opts.levelModule)) {
+      consoleTransport.setModuleLevel(module, level);
+    }
+  }
+
+  const transports: TransportStream[] = [consoleTransport];
+
+  // yargs populates with undefined if just set but with no arg
+  // $ ./bin/lodestar.js beacon --logFileDailyRotate
+  // args = {
+  //   logFileDailyRotate: undefined,
+  // }
+  // `lodestar --logFileDailyRotate` -> enabled daily rotate with default value
+  // `lodestar --logFileDailyRotate 10` -> set daily rotate to custom value 10
+  // `lodestar --logFileDailyRotate 0` -> disable daily rotate and accumulate in same file
+  if (opts.file) {
+    const filename = opts.file.filepath;
+
+    transports.push(
+      opts.file.dailyRotate != null
+        ? new DailyRotateFile({
+            level: opts.file.level,
+            //insert the date pattern in filename before the file extension.
+            filename: filename.replace(/\.(?=[^.]*$)|$/, "-%DATE%$&"),
+            datePattern: DATE_PATTERN,
+            handleExceptions: true,
+            maxFiles: opts.file.dailyRotate,
+            auditFile: path.join(path.dirname(filename), ".log_rotate_audit.json"),
+          })
+        : new winston.transports.File({
+            level: opts.file.level,
+            filename: filename,
+            handleExceptions: true,
+          })
+    );
+  }
+
+  return transports;
+}
+
+interface DefaultMeta {
+  module: string;
+}
+
+export class WinstonLoggerNode extends WinstonLogger implements LoggerNode {
+  constructor(private readonly opts: LoggerNodeOpts, private readonly transports: winston.transport[]) {
+    const defaultMeta: DefaultMeta = {module: opts?.module || ""};
+    super(
+      winston.createLogger({
+        // Do not set level at the logger level. Always control by Transport, unless for testLogger
+        level: opts.level,
+        defaultMeta,
+        format: getFormat(opts),
+        transports,
+        exitOnError: false,
+        levels: logLevelNum,
+      })
+    );
+  }
+
+  static fromNewTransports(opts: LoggerNodeOpts): WinstonLoggerNode {
+    return new WinstonLoggerNode(opts, getNodeLoggerTransports(opts));
+  }
+
+  // Return a new logger instance with different module and log level
+  // but a reference to the same transports, such that there's only one
+  // transport instance per tree of child loggers
+  child(opts: LoggerNodeChildOpts): LoggerNode {
```

Comment:
@dapplion This changed how we create child loggers, it creates a new event listener for each child logger
- https://github.com/ChainSafe/lodestar/issues/5529

Was the previous implementation not compatible when using workers?
https://github.com/ChainSafe/lodestar/blob/375d660ed1e7ba13945dff41e63154a26c4be008/packages/utils/src/logger/winston.ts#L82

Using native `winston.child` does not work as mention in previous comment.

Workaround for now is to just increase max listeners, are we fine with that approach for now?
- https://github.com/ChainSafe/lodestar/pull/5537

Reply:
Hmm unsure of the reason forgot to document. Could you test if the previous approach works fine still?

Reply:
child logger are only created in the main thread, does not affect workers at all because we initialize a completely new logger instance there https://github.com/ChainSafe/lodestar/blob/e2e5417f48cac848cc0eb0b34389c9a6b7673444/packages/beacon-node/src/network/core/networkCoreWorker.ts#L38

Opened a PR to revert `logger.child` implementation back to original approach
- https://github.com/ChainSafe/lodestar/pull/5597